### ISSUE TITLE
REST API filtering/set_url broken

### ISSUE THREAD

**Issue by bertabus** [AdguardTeam/AdGuardHome#2243]
### Issue Details

* **Version of AdGuard Home server:**
  * Version: v0.104.0
* **How did you setup DNS configuration:**
  * adguard running on local server, DNS assigned by router to point to local server
* **If it's a router or IoT, please write device model:**
  * amd64 server
* **Operating system and version:**
  * Docker install with http port forwarded from 3000 to 880

### Expected Behavior
```sh
curl -X POST "http://192.168.1.6:880/control/filtering/set_url" -H  "accept: */*" -H  "Content-Type: application/json" -d "{\"url\":\"https://raw.githubusercontent.com/DandelionSprout/adfilt/master/GameConsoleAdblockList.txt\",\"enabled\":false}" -u user:pass
```
should return an OK or similar 

### Actual Behavior
returns
```sh
invalid URL or file path
```

### Additional filtering issue with remove_url, but not as major
running
```sh
curl -X POST "http://192.168.1.6:880/control/filtering/add_url" -H  "accept: */*" -H  "Content-Type: application/json" -d "{\"name\":\"testURL\",\"url\":\"https://raw.githubusercontent.com/DandelionSprout/adfilt/master/GameConsoleAdblockList.txt\"}" -u user:pass
```
returns
`OK adding 9 rules` as expected

running
```sh
curl -X POST "http://192.168.1.6:880/control/filtering/remove_url" -H  "accept: */*" -H  "Content-Type: application/json" -d "{\"name\":\"testURL\",\"url\":\"https://raw.githubusercontent.com/DandelionSprout/adfilt/master/GameConsoleAdblockList.txt\"}" -u user:pass
```
returns nothing which causes errors to flag in upstream repos (notably hassio addon) that are expecting an "OK" returned. Although the call does in fact remove the url as expected.

**Comment by ainar-g** [comment]
<p>Good&#xA0;day!&#x2003;Thanks for&#xA0;filing this issue.&#x2003;Seems
like&#xA0;a&#xA0;documentation error, because&#xA0;the&#xA0;API docs describe
the&#xA0;wrong schema.&#x2003;This API is supposed to&#xA0;update
the&#xA0;properties of&#xA0;a&#xA0;previously set filter.&#x2003;Please create
a&#xA0;filter, unless&#xA0;you already did, and&#xA0;try
the&#xA0;following&#xA0;command:</p>

<pre>$ curl \
-H "Accept: */*" \
-H "Content-Type: application/json" \
-X POST \
-d '{"data":{"enabled":false,"name":"YOUR_NAME_HERE","url":"YOUR_URL_HERE"},"url":"YOUR_URL_HERE","whitelist":false}' \
-u user:pass \
'http://192.168.1.6:880/control/filtering/set_url'
</pre>

<p>And&#xA0;set the&#xA0;<code>enabled</code> parameter and&#xA0;other
parameters as&#xA0;you&#xA0;need.</p>

**Comment by bertabus** [comment]
Good day, thanks for taking a look.

That appears to be it, I tried again using your above JSON schema.
Following results using your schema above and trying the different filtering controls:

* **control/filtering/add_url:** returns "OK 9 rules" 
  - Shows up in my dns blocklist and is also enabled (even though my json said disabled, not a big deal but something to note.)
* **control/filtering/set_url:** returns ""
  - setting json "enabled" works as expected in enabling the blocklist.
  - setting json "disabled" works as expected in disabling the blocklist.
* **control/filtering/remove_url:** returns ""
  - Works as expected in that the URL is removed from my DNS blocklist

So it looks like indeed the API documentation is wrong. 

The other potential issue is the feedback inconsistencies returned on the call. Is there any expectation here?

**Comment by ainar-g** [comment]
<p>Thanks&#xA0;for&#xA0;the&#xA0;information!&#x2003;We will soon fix
the&#xA0;<code>add_url</code> documentation (there is
no&#xA0;<code>enabled</code> field there) and&#xA0;the&#xA0;response
from&#xA0;<code>remove_url</code>.&#x2003;I'm&#xA0;not&#xA0;entirely sure what&#xA0;you mean by&#xA0;“feedback
inconsistencies”, but&#xA0;we will strive to&#xA0;make the&#xA0;API more uniform
in&#xA0;the&#xA0;following&#xA0;releases.</p>

**Comment by ainar-g** [comment]
<p>All&#xA0;issues should be fixed as&#xA0;of 02d16a0.&#x2003;Please let us know
if&#xA0;something is still&#xA0;missing.</p>

**Comment by bertabus** [comment]
All I meant by "feedback inconsistencies" was that the API returns OK in one case and not in others.
Just checked the API doc's and everything looks good, I appreciate the clarification.
Thanks again for the update.
